pandas/core/indexing.py

```diff
@@ -429,11 +430,23 @@ def _setitem_with_indexer(self, indexer, value):
                     # this preserves dtype of the value
                     new_values = Series([value])._values
                     if len(self.obj._values):
-                        try:
-                            new_values = np.concatenate([self.obj._values, new_values])
-                        except TypeError:
+                        if is_timedelta64_dtype(
```

Comment:
I think you can avoid all of this trouble (and blow away almost all of this code) by using ``_concat_compat`` here instead.

Reply:
will take a look

Reply:
Doing this breaks two tests, both of them taking an empty Series and setting an integer value.  Using _concat_compat, we apparently infer float64 instead of int64

```
    def test_partial_set_empty_series(self):
    
        # GH5226
    
        # partially set with an empty object series
        s = Series()
        s.loc[1] = 1
>       tm.assert_series_equal(s, Series([1], index=[1]))
E       AssertionError: Attributes are different
E       
E       Attribute "dtype" are different
E       [left]:  float64
E       [right]: int64
```
```

Reply:
never mind.  pending finishing the test suite locally, this looks like it works